src/plugins/enzymes/src/FindEnzymesDialog.cpp

```diff
@@ -273,6 +288,15 @@ void EnzymesSelectorWidget::setEnzymesList(const QList<SEnzymeData>& enzymes) {
         EnzymeGroupTreeItem* gi = dynamic_cast<EnzymeGroupTreeItem*>(item);
         if (ei != nullptr) {
             teSelectedEnzymeInfo->setHtml(ei->getEnzymeInfo());
+            if (!ei->hasNumberCalculationTask && advSequenceContext != nullptr) {
+                auto seqObj = advSequenceContext->getSequenceObject();
+                const auto& er = seqObj->getEntityRef();
+                U2Region reg(0, seqObj->getSequenceLength());
+                auto t = new FindSingleEnzymeTask(er, reg, ei->enzyme, nullptr, seqObj->isCircular());
```

Comment:
Do not run the task for very large sequences. Info like: "the sequence has 782783723 hits" is a garbage info.
Limit the run to sequences ~10k

Reply:
The dependency of restriction sites quantity from sequence length is quite implicit, I would even say, that there is almost no such dependency at all. For example, on human_T1 some enzymes occurs thousand times and some other - just several times or even zero, so this limitation could be very rough and it means, that we just forbid to find restriction enzymes in sequences larger than 10'000, which is bad (I didn't see such limitations is some other bioinformatic tools). 

It you want to add some limitation (but, as I think, this have no real sense, because all calculations passes very fast and sequences of any size) I can limit calculation of each enzyme, for example, 10'000 enzymes max. If there is more enzymes that 10'000 I will write "more than 10'000 sites".

Reply:
I added a maximum results limit - 10'000. `FindSingleEnzymeTask` just stops if there are more sites than that number. Calculation of 10'000 sites never require more than 1-2 seconds

Reply:
I generated a random sequence of size 100Mb and already have a lot of long running tasks (each runs ~10 seconds on my computer):

![image](https://github.com/ugeneunipro/ugene/assets/17791039/b9bda9e0-7c53-4987-9912-e09c4207570f)

What will be with a larger sequence size? The tasks have high pressure to memory, CPU, IO and are not canceled: so we have a multiplied pressure.

Reply:
I limited sequence length 200'000 bases max, I should be good, because on human"T1 everything works perfectly